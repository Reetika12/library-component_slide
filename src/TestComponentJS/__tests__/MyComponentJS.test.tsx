import * as React from 'react';
import { act } from 'react-dom/test-utils';
import * as ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import MyComponent from '../index';

describe('MyComponent', function test() {
	it('should render correctly', () => {
		const component = shallow(<MyComponent title="1" />);

		expect(component).toMatchSnapshot();
	});

	it('should display title passed as props', function () {
		const title = 'hello';
		const container = document.createElement('div');
		document.body.appendChild(container);
		act(() => {
			ReactDOM.render(<MyComponent title={title} />, container);
		});
		const header = container.querySelector('h1');
		expect(header?.textContent).toBe(title);
	});
});
