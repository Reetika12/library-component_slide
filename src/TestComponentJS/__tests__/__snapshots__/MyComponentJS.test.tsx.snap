// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly 1`] = `
<div
  className="test-component-js"
>
  <h1>
    1
  </h1>
  <p>
    This is an example component.
  </p>
</div>
`;
